org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/Document.xtend

```diff
@@ -20,8 +20,19 @@ import org.eclipse.lsp4j.Range
 
     Integer version
     String contents
+    boolean printSourceOnError
```

Comment:
am not sure if we need explicit getters with since for the flag too

Reply:
I can see why we would want that, but it's quite annoying to add those in an @Data class

Reply:
yes the same is for contructors. @szarnekow what is our general approach regarding api and active annotation produced code

Reply:
Usually one has to resort to manually written code to get the annotations right and sound.